homeassistant/components/vallox/sensor.py

```diff
@@ -68,7 +68,7 @@ class ValloxProfileSensor(ValloxSensor):
     """Child class for profile reporting."""
 
     @property
-    def native_value(self) -> StateType:
+    def native_value(self) -> StateType | datetime:
```

Comment:
Is this one needed?

My impression is this is valid, and style wise you can argue its more accurate, but i don't claim to be an expert in typing best practices.
```
class A(ABC):
    def example(self) -> str | bool:
        """Some example."""

class B(A):
    def example(self) -> str:
        return ""
```